core/src/Revolution/Processors/Security/User/Create.php

```diff
@@ -232,6 +232,52 @@ public function sendNotificationEmail() {
                 'html' => true,
             ]);
         }
+
+        if ($this->getProperty('passwordgenmethod') === 'user_email_specify') {
+            $activationHash = md5(uniqid(md5($this->object->get('email') . '/' . $this->object->get('id')), true));
```

Comment:
I propose to use the `hashing` service to generate the hash. 

For reference: https://github.com/modxcms/revolution/blob/2ee2c45cda11e578595990874407fbfa668f4ff4/core/src/Revolution/modUser.php#L67